drivers/sensor/als31300/als31300.c

```diff
@@ -0,0 +1,559 @@
+ * @param raw_temp 12-bit raw temperature value
+ * @param q31_out Pointer to store Q31 value
+ */
+static void als31300_convert_temp_to_q31(uint16_t raw_temp, q31_t *q31_out)
+{
+	/* Convert to microcelsius using integer arithmetic */
+	int32_t microcelsius = als31300_convert_temperature(raw_temp);
+
+	/* Convert to Q31 format: Q31 = (value * 2^shift) / 1000000
+	 * For temperature, we use shift=16, so the full scale is ±2^(31-16) = ±32768°C
+	 * This gives us good resolution for typical temperature ranges (-40°C to +125°C)
+	 * microcelsius * 2^16 / 1000000 = microcelsius * 65536 / 1000000
+	 */
+	*q31_out = (q31_t)(((int64_t)microcelsius << ALS31300_TEMP_SHIFT) / 1000000);
+}
+
+/**
+ * @brief RTIO synchronous submit function
+ */
+static void als31300_submit_sync(struct rtio_iodev_sqe *iodev_sqe)
+{
+	const struct sensor_read_config *cfg = iodev_sqe->sqe.iodev->data;
+	const struct device *dev = cfg->sensor;
+	const struct sensor_chan_spec *channels = cfg->channels;
+	const size_t num_channels = cfg->count;
+
+	uint32_t min_buf_len = sizeof(struct als31300_encoded_data);
+	uint8_t *buf;
+	uint32_t buf_len;
+	uint64_t cycles;
+
+	/* Get buffer for encoded data */
+	int rc = rtio_sqe_rx_buf(iodev_sqe, min_buf_len, min_buf_len, &buf, &buf_len);
+
+	if (rc != 0) {
+		LOG_ERR("RTIO buffer allocation failed: %d", rc);
+		rtio_iodev_sqe_err(iodev_sqe, rc);
+		return;
+	}
+
+	/* Get timestamp */
+	rc = sensor_clock_get_cycles(&cycles);
+	if (rc != 0) {
+		LOG_ERR("Failed to get timestamp: %d", rc);
+		rtio_iodev_sqe_err(iodev_sqe, rc);
+		return;
+	}
+
+	struct als31300_encoded_data *edata = (struct als31300_encoded_data *)buf;
+
+	edata->header.timestamp = sensor_clock_cycles_to_ns(cycles);
+
+	/* Set channel flags based on requested channels */
+	edata->has_magn_x = 0;
+	edata->has_magn_y = 0;
+	edata->has_magn_z = 0;
+	edata->has_temp = 0;
+
+	for (size_t i = 0; i < num_channels; i++) {
+		switch (channels[i].chan_type) {
+		case SENSOR_CHAN_MAGN_X:
+			edata->has_magn_x = 1;
+			break;
+		case SENSOR_CHAN_MAGN_Y:
+			edata->has_magn_y = 1;
+			break;
+		case SENSOR_CHAN_MAGN_Z:
+			edata->has_magn_z = 1;
+			break;
+		case SENSOR_CHAN_MAGN_XYZ:
+			edata->has_magn_x = 1;
+			edata->has_magn_y = 1;
+			edata->has_magn_z = 1;
+			break;
+		case SENSOR_CHAN_AMBIENT_TEMP:
+			edata->has_temp = 1;
+			break;
+		case SENSOR_CHAN_ALL:
+			edata->has_magn_x = 1;
+			edata->has_magn_y = 1;
+			edata->has_magn_z = 1;
+			edata->has_temp = 1;
+			break;
+		}
+	}
+
+	/* Fetch sensor data */
+	rc = als31300_read_sensor_data(dev, SENSOR_CHAN_ALL, &edata->readings);
+	if (rc != 0) {
+		LOG_ERR("Failed to fetch sensor data: %d", rc);
+		rtio_iodev_sqe_err(iodev_sqe, rc);
+		return;
+	}
+
+	rtio_iodev_sqe_ok(iodev_sqe, 0);
+}
+
+/**
+ * @brief RTIO submit function
+ */
+static void als31300_submit(const struct device *dev, struct rtio_iodev_sqe *iodev_sqe)
+{
+	struct rtio_work_req *req = rtio_work_req_alloc();
```

Comment:
I'd suggest staying away from using the RTIO workqueue unless we need to. In this case, a set of SQE's coordinating an async read, finalized by a cb should be enough. Here's a reference implementing this: https://github.com/zephyrproject-rtos/zephyr/blob/ba65ff34fbd4a46a6d8e5cb20bb30a8cacdeb429/drivers/sensor/bosch/bmp581/bmp581.c#L606-L680

Reply:
I addressed this to follow the same pattern like bmp581 does